### ISSUE TITLE
Add flag to return cache directory

### ISSUE THREAD

**Issue by dbrgn** [rust-lang/cargo#5412]
Since the location of the cache directory depends on various factors (OS, env vars etc) it would be great to have a CLI flag that would return the path to the currently used cache directory.

This would be especially useful in CI.

**Comment by matklad** [comment]
Might be addressed by https://github.com/rust-lang/cargo/pull/5183

**Comment by dbrgn** [comment]
@alexcrichton I meant the ~/.cargo directory. AFAIK it can be changed with the CARGO_HOME env var.

**Comment by alexcrichton** [comment]
Indeed, and are you setting that env var on CI?

**Comment by dbrgn** [comment]
I am now. The thing is that some docker images also set that variable, so if I'm not setting the variable explicitly, it's not immediately clear where the cache directory is located. Especially since the location might change in the future.

But it seems that this will be addressed by #5183 indeed :slightly_smiling_face:

**Comment by soc** [comment]
@dbrgn Yes, that's one of the use cases I have! After things have progressed a bit in #5183, I would appreciate if you gave it a try and told me whether everything works for your use case.

**Comment by dwijnand** [comment]
Would adding a `dirs` subcommand, with no args, that prints a handful of directory paths (without changing what those paths are today) be a welcome PR?  Or would an RFC for it be required first?

**Comment by alexcrichton** [comment]
@dwijnand I think we could probably do that yeah, but I suspect we should hold off in the near term (with the soft feature freeze and whatnot) and we likely want to land it in tandem or in closer coordination with the PR that switches around cargo's caches

**Comment by dwijnand** [comment]
Sounds good!

**Comment by soc** [comment]
@alexcrichton From my perspective it would actually be fine to get this done earlier than the rest of #5183, because it would help debugging the various changes there; generally cutting out as many easy parts as possible would help focusing on the harder parts later.

Oh, and btw, welcome @dwijnand to the cargo team! Looks like we are getting a full house of Scala refugees lately. ;-)

(I think it would even be fine supporting individual sub-commands, even if they aren't particularly helpful at the moment. It would allow tools that rely on specific directories to prepare early for a possible migration.)
